BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginFormState.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.biometricloginsample
+
+sealed class LoginFormState
```

Comment:
```
class LoginState {
  data class Failed
  data class Successful
}
```
is preferred here

Reply:
LoginFormState vs LoginState: there are two login paths. This class only tracks the form not the biometric path.

Also would you please explain why sealed class is not good here? Thanks

Reply:
Ah, I didn't see the two states. Sealed is okay here.
Above mentioned pattern is being used throughout other APIs as well.